### ISSUE TITLE
Do a better job of guiding users that accidentally open parent folders of Flutter projects

### ISSUE THREAD

**Issue by cannoneyed** [Dart-Code/Dart-Code#1146]
When launching vscode without an ios simulator open, the flutter tools behave in a way that is unexpected and is particularly challenging for beginners to understand.

There's no indication in the vscode lower toolbar that no devices are unavailable, and only the following commands are available:

```
Run Flutter Doctor
Run Flutter Upgrade
New Project
```

When selecting `Start Debugging`, flutter tries to build to the stand-alone vm which causes the following error while running through the tutorials:

```
The built-in library 'dart:ui' is not available on the stand-alone VM.
```

I think this would be pretty easily fixed by indicating in the bottom-right that no devices are available (where the normal device list is). It's also extremely frustrating that the `Flutter: Select Device` and `Flutter: Launch Emulator` commands aren't available when no emulator / device is running / connected.

**Comment by cannoneyed** [comment]
Yup, I see now - that's exactly what's going on here. The issues I saw were the result of opening the project's _parent_ directory in VSCode. 

Basically, I created a directory, opened it in VSCode, created a new project using the `New Project` command (which created a **new** project directory inside of that, then opened up a new VSCode window for that directory). Everything worked fine there. When later opening VSCode in the _original_ directory, I ran into the error. 

Anyway, this totally makes sense in hindsight... Perhaps the only thing that might be worth thinking about is if the `Debug: Start Debugging` command knows to look "up" a directory to try to run the dart project, then maybe the Flutter tools might want to look "up" as well? Although I can understand why this might be tricky.

Thanks for the quick response!

**Comment by DanTup** [comment]
This behaviour still kinda bugs me. We did previously scan your folder and if we found Flutter projects in sub-folders, we prompted you to upgrade them to "workspace folders" (this is a way of having multiple project folders open, and as long as one of them is Flutter, we'll go into Flutter mode).

However, this was kinda annoying and we ended up removing it because the way VS Code represents these when they're in the same tree is bad.

The problem with just going into Flutter mode and trying to do our best, is that when you hit F5, if your open folder isn't a Flutter project directly, we don't really know where to invoke Flutter (or what entry point to use). By forcing the user to either open the Flutter project directly, or at least use workspace folders, the UI is clear which folder is being launched.

I'm gonna re-open this as a reminder that people still hit this issue, but I'm not sure there's an obvious solution right now.

**Comment by DanTup** [comment]
vNext will look in the first level of folders for a Flutter project and if found, go into Flutter-mode. I believe that'll solve this issue (as long as you have a Flutter project in the first level of folders - we don't walk down the entire tree). The previous issue about pressing F5 not doing the right thing was solved as part of #1075.

I'll try and make a beta build this week for testing.

**Comment by DanTup** [comment]
Beta here if you'd like to test it before it's released -> https://github.com/Dart-Code/Dart-Code/releases/tag/v2.20.0-beta.1
